state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -790,16 +811,12 @@ pub mod pallet {
 				}
 			}
 
-			let mut expired_transactions = Vec::new();
-
-			for (account, tx_id, expire_block) in TaintedTransactions::<T, I>::iter() {
-				if expire_block <= now {
-					expired_transactions.push((account.clone(), tx_id.clone()));
+			for (account, tx_id) in ReportExpiresAt::<T, I>::take(now) {
```

Comment:
Can we add a code comment to explain what this is for? It took me a bit of time to understand the motivation here. Also, are we ever removing `TaintedTransactions` record if we prewitnessed a deposit? I couldn't find it. I think we could remove it once the channel expires (which is aslo when we forget the "boosted" status).

Reply:
We remove it when it's finally witnessed. The idea behind the code is that for some reason the timespan between pre witness and witness is so long that it expires in the meantime, we have to add it back since otherwise the boosted swap would go through. Eventually it will get removed from `TaintedTransactions ` though.

Reply:
> Eventually it will get removed from TaintedTransactions  though.

Does this include the case where a a tx is prewitnessed, but is never fully witnessed? (This is the case when boosters lose money essentially.)